src/ctia/import/threatgrid/ttps.clj

```diff
@@ -0,0 +1,78 @@
+(ns ctia.import.threatgrid.ttps
+  (:require [clojure.data.json :as json]
+            [cheshire.core :as cjson]
+            [clojure.java.io :as io]
+            [clojure.pprint :refer [pprint]]
+            [ctim.schemas.ttp :as ttp]
+            [schema.core :as s]
+            [ctia.import.threatgrid.http :as http]
+            [ctia.lib.time :as time]))
+
+(s/defschema TGTTP
+  {(s/required-key "id") s/Str
+   (s/required-key "title") s/Str
+   (s/required-key "description") s/Str
+   ;;(s/required-key "short_description") s/Str
+   (s/required-key "timestamp") s/Str
+   (s/required-key "type") s/Str
+   (s/required-key "expires") s/Str
+   (s/required-key "indicators") [(s/maybe s/Str)]})
+
+(defn tg-ttp->ctim-ttp [validate-tg-ttp-fn validate-ctim-ttp-fn]
+  (fn [{:strs [id title description timestamp type expires indicators]
+        :as tg-ttp}]
+    (validate-tg-ttp-fn tg-ttp)
+    (validate-ctim-ttp-fn
+     (-> {:title id
+          :description description
+          :short_description description
+          :tlp "green"
+          :valid_time (time/date-str->valid-time timestamp :days 30)
+          ;; :intended_effect ""
+          ;; :behavior ""
+          ;; :resources ""
+          ;; :victim_targeting ""
+          ;; :exploit_targets ""
+          ;; :related_ttps ""
+          :indicators indicators
+          :ttp_type type
+          :source "Threat Grid"}))))
+
+(defn tg-ttps->ctim-ttps
+  [ttps & {:keys [validate?]
+           :or {validate? false}}]
+  (let [validate-tg-ttp (if validate?
+                          (partial s/validate TGTTP)
+                          identity)
+        validate-ctim-ttp (if validate?
+                            (partial s/validate ttp/NewTTP)
+                            identity)
+        transform (tg-ttp->ctim-ttp validate-tg-ttp validate-ctim-ttp)]
+    (for [tg-ttp ttps]
+      (transform tg-ttp))))
```

Comment:
I believe it is more common to use

~~~clojure
(map transform ttps)
~~~

instead of a for here.